docs/source-2.0/additional-specs/smoke-tests.rst

```diff
@@ -0,0 +1,194 @@
+    ``list`` of :ref:`SmokeTestCase <SmokeTestCase-struct>` structures
+
+The ``smokeTests`` trait is a list of :ref:`SmokeTestCase <SmokeTestCase-struct>` structures.
+
+.. _SmokeTestCase-struct:
+
+SmokeTestCase
+=============
+
+A structure defining a single smoke test case.
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 25 65
+
+    * - Property
+      - Type
+      - Description
+    * - id
+      - ``string``
+      - **Required**. The identifier of the test case. This identifier may be
+        used by smoke test implementations to generate test case names. The
+        provided ``id`` MUST match Smithy's :token:`smithy:Identifier` ABNF. No
+        two test cases can share the same ID, including test cases defined for
+        other operations bound to the same service.
+    * - params
+      - ``document``
+      - Defines the input parameters used to generate the request. These
+        parameters MUST be compatible with the input of the operation.
+
+        Parameter values that contain binary data MUST be defined using values
+        that can be represented in plain text as the plain text representation
+        (for example, use "foo" and not "Zm9vCg==").
+    * - vendorParams
+      - ``document``
+      - Defines vendor-specific parameters that are used to influence the
+        request. For example, some vendors might utilize environment variables,
+        configuration files on disk, or other means to influence the
+        serialization formats used by clients or servers.
+
+        If a ``vendorParamsShape`` is set, these parameters MUST be compatible
+        with that shape's definition.
+    * - vendorParamsShape
+      - ``string``
+      - A shape to be used to validate the ``vendorParams`` member contents.
+
+        If set, the parameters in ``vendorParams`` MUST be compatible with this
+        shape's definition.
+    * - expect
+      - :ref:`Expectation <Expectation-union>`
+      - **Required**. Defines the response that is expected from the service
+        call. This can be either a successful response, an error message, or
+        a specific error response.
+    * - tags
+      - ``[string]``
+      - Attaches a list of tags that can be used to categorize and group
+        test cases. If a test case uses a feature that requires special
+        configuration, it should be tagged.
+
+.. _Expectation-union:
+
+Expectation
+-----------
+
+A union describing the different kinds of expectations that can be made for a
+test case. As it is a union, exactly one member must be set.
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 25 65
+
+    * - Property
+      - Type
+      - Description
+    * - success
+      - ``Unit``
+      - Indicates that the call is expected to not throw an error. No other
+        assertions are made about the response.
+    * - failure
+      - :ref:`FailureExpectation <FailureExpectation-struct>`
+      - Indicates that the call is expected to throw an error.
```

Comment:
nit: not in the scope of this PR, but do we want to mark `Expectation` as `@unstable`?
I feel like there could be different types of expectations we could add in the future.